pandas/core/frame.py

```diff
@@ -1557,6 +1557,20 @@ def to_records(self, index=True, convert_datetime64=None):
 
             Whether to convert the index to datetime.datetime if it is a
             DatetimeIndex.
+        column_dtypes : str, type, dict, default None
+            .. versionadded:: 0.24.0
+
+            If a string or type, the data type to store all columns. If
```

Comment:
why is this anything but a dict of column_name -> dtype? what does the indices have to do with this?

Reply:
You can map column indices to dtypes e.g.

```
{0: "int32"}
```

maps `df.columns[0]` to the `int32` `dtype`.

This design was taken from `read_csv`, where we do something similar.

Reply:
> are you testing for a multi-index for the index? I think this is ok
multi-index on the columns? does a rec-array even work with this?

Reply:
Added tests for MI, both in the columns and indices.